### PR TITLE
fix: add Redis optional configurable password

### PR DESCRIPTION
**Pull request by ljobse**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
   Added a password option to the redis instance

- **Why was this change needed?** (You can also link to an open issue here)
  Currently it's not possible to connect with an external redis instance

- **Other information**:
  Related to: https://github.com/novuhq/novu/issues/1863

### CODE REVIEW

File: apps/api/src/app/events/services/workflow.queue.service.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export class WorkflowQueueService {
       db: Number(process.env.REDIS_DB_INDEX),
       port: Number(process.env.REDIS_PORT),
       host: process.env.REDIS_HOST,
+      password: process.env.REDIS_PASSWORD,
```
Comment (p-fernandez) [inline apps/api/src/app/events/services/workflow.queue.service.ts:15]: _For the reader's convenience..._

BullMQ's `QueueBaseOptions` (https://github.com/taskforcesh/bullmq/blob/89bad097edf1692dba19eec938da82b6f2be7430/src/interfaces/redis-options.ts)
is extending `IORedis.RedisOptions` that is defined here:
(https://github.com/luin/ioredis/blob/eabb70ed835828ad85b602a217992486e05d2218/lib/cluster/util.ts)

File: docs/docs/community/run-locally.md [hunk 2]
```diff
@@ -57,6 +57,7 @@ If you want to test certain parts of Novu or run it in production mode though, s
         <li><code>DISABLE_USER_REGISTRATION</code> (default: false)<br />If users should not be able to create new accounts. Possible values are: true, false</li>
         <li><code>REDIS_HOST</code><br />The domain / IP of your redis instance</li>
         <li><code>REDIS_PORT</code><br />The port of your redis instance</li>
+        <li><code>REDIS_PASSWORD</code><br />Optional password of your redis instance</li>
```
Comment (p-fernandez) [inline docs/docs/community/run-locally.md:60]: Thanks for this. ❤️

**Review by p-fernandez — APPROVED** [review]
🌟 

I was thinking in suggesting to add the new environment variable in the example dot env files but Redis throws an error message when connecting with an empty password, so with the documentation should be enough to guide users.

**Comment by ljobse** [comment]
@p-fernandez I see a step to assign a milestone is failing, should I take any action to resolve this?

**Comment by p-fernandez** [comment]
> @p-fernandez I see a step to assign a milestone is failing, should I take any action to resolve this?

It is a matter of our CI. So nothing on your side. Thank you.

**Pull request merged**